api/all/src/main/java/io/opentelemetry/api/trace/TraceFlags.java

```diff
@@ -44,6 +44,28 @@ static TraceFlags getSampled() {
     return ImmutableTraceFlags.SAMPLED;
   }
 
+  /**
+   * Returns the int (fixed32) representation of the 4 bytes flags with the
+   * has_parent_context_is_remote flag bit on.
+   *
+   * @return the int (fixed32) representation of the 4 bytes flags with the *
+   *     has_parent_context_is_remote flag bit on.
+   */
+  static int getHasParentIsRemote() {
```

Comment:
The way that the proto is arranged, SpanFlags is composed os 8 bits for TraceFlags, with the remaining bits allocated for other purposes. 

I think extending `TraceFlags` with these methods muddies the waters. Let's keep TraceFlags mirroring the proto representation, and instead create a new `SpanFlags` class, and include these utility methods in it:
```
public final class SpanFlags {

  private static final int CONTEXT_HAS_IS_REMOTE_BIT = 0x00000100;
  private static final int CONTEXT_IS_REMOTE_BIT = 0x00000200;
  private static final int CONTEXT_IS_REMOTE_MASK = CONTEXT_HAS_IS_REMOTE_BIT | CONTEXT_IS_REMOTE_BIT;

  private SpanFlags() {}

  /**
   * Returns the int (fixed32) representation of this {@link TraceFlags} enriched with the flags
   * indicating a remote parent.
   *
   * @param isParentRemote indicates whether the parent context is remote
   * @return the int (fixed32) representation of this {@link TraceFlags} enriched with the flags
   *     indicating a remote parent.
   */
  public static int withParentIsRemoteFlags(TraceFlags traceFlags, boolean isParentRemote) {
    byte byteRep = traceFlags.asByte();
    if (isParentRemote) {
      return (byteRep & 0xff) | CONTEXT_IS_REMOTE_MASK;
    }
    return (byteRep & 0xff) | CONTEXT_HAS_IS_REMOTE_BIT;
  }

  /**
   * Returns the int (fixed32) representation of the 4 bytes flags with the
   * has_parent_context_is_remote flag bit on.
   *
   * @return the int (fixed32) representation of the 4 bytes flags with the *
   *     has_parent_context_is_remote flag bit on.
   */
  public static int getHasParentIsRemoteMask() {
    return ImmutableTraceFlags.CONTEXT_HAS_IS_REMOTE_BIT;
  }

  /**
   * Returns the int (fixed32) representation of the 4 bytes flags with the
   * has_parent_context_is_remote and parent_context_is_remote flag bits on.
   *
   * @return the int (fixed32) representation of the 4 bytes flags with the
   *     has_parent_context_is_remote and parent_context_is_remote flag bits on.
   */
  public static int getParentIsRemoteMask() {
    return ImmutableTraceFlags.CONTEXT_IS_REMOTE_MASK;
  }
}
```
Could also provide a helper method for extracting `TraceFlags` from the span flags `int`. 

Note I've also renamed `getHasParentIsRemote`, `getParentIsRemote` to `getHasParentIsRemoteMask`, `getParentIsRemoteMask`. I found the method names confusing and implying that they somehow returned the boolean status of whether the parent is remote. 

Should also include an example usage like included in the proto comments:
```
  // To read whether the value is known, use `(flags & SPAN_FLAGS_CONTEXT_HAS_IS_REMOTE_MASK) != 0`.
  // To read whether the span is remote, use `(flags & SPAN_FLAGS_CONTEXT_IS_REMOTE_MASK) != 0`.
```

Reply:
> I think extending TraceFlags with these methods muddies the waters. Let's keep TraceFlags mirroring the proto representation, and instead create a new SpanFlags class, and include these utility methods in it:

That was also what I was thinking and sort of what I implemented initially. Changed it based on https://github.com/open-telemetry/opentelemetry-java/pull/6388#discussion_r1568019556. Will adapt based on your proposal.